src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -13,6 +13,40 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
 {
     internal sealed class SourcePropertySymbol : SourcePropertySymbolBase
     {
+        internal bool ContainsFieldKeyword
+        {
+            get
+            {
+                if (CSharpSyntaxNode is not PropertyDeclarationSyntax property)
+                {
+                    return false;
+                }
+
+                if (property.AccessorList is not { } accessorList)
+                {
+                    return false;
+                }
+
+                foreach (var accessor in accessorList.Accessors)
+                {
+                    var body = ((SyntaxNode?)accessor.Body ?? accessor.ExpressionBody);
```

Comment:
>((SyntaxNode?)accessor.Body ?? accessor.ExpressionBody [](http://example.com/codeflow?start=31&length=54)

Do we need to look at both? I don't remember if SemanticModel can bind either body.

Reply:
@AlekseyTs I don't get how this relates to SemanticModel. We're looking for "field" keyword. So we want to look for `get { return field; }` and also `get => field;`

Reply:
>>Do we need to look at both? I don't remember if SemanticModel can bind either body.

> I don't get how this relates to SemanticModel. We're looking for "field" keyword. So we want to look for ```get { return field; }``` and also ```get => field;```

I do not remember if SemanticModel can bind an expression body in presence of a block body. If it can, then we might need to look for the keyword in both bodies, not just one. Regardless of the current behavior, we need tests specifically covering this situation. I would add a test that asserts result of GetSymbolInfo for "field" keywords when:
- both block body and expression body use the "field" keyword. Test scenario when we call GetSymbolInfo for the "field" keyword in block body first and test scenario when we call GetSymbolInfo for the "field" keyword in expression body first.
- expression body uses the "field" keyword, but block body doesn't
- block body uses the "field" keyword, but expression body doesn't

Reply:
@AlekseyTs Is anything here still relevant?

Reply:
> Is anything here still relevant?

Yes. we need to add suggested tests eventually. Capturing this in a prototype comment should be good enough for the initial PR

Reply:
@AlekseyTs There are couple of tests around this I believe. `TestSemanticModel_GetterOnly`, `TestSemanticModel_GetterAndSetter_BothAreExpressionBodied`, `TestSemanticModel_GetterAndSetter_MixBodyAndExpressionBoded`, `TestSemanticModel_GetterAndSetter_OnlyOneAccessorUsesKeyword`.

Reply:
> There are couple of tests around this I believe.

I think those do not quite cover the aspect that is "interesting" in this context. We should have tests confirming that SemanticModel doesn't bind an expression body in presence of a block body.

Reply:
@AlekseyTs Sorry, I don't quite get the exact scenario you're referring to, nor what SemanticModel API(s) we want to test. Can you elaborate more please?

Reply:
> Sorry, I don't quite get the exact scenario you're referring to, nor what SemanticModel API(s) we want to test. Can you elaborate more please?

In an error scenario, a property accessor can have both an expression body and a block body. In this case we are ignoring the expression body for the purpose of the syntactic ```field``` check. So, "We should have tests confirming that SemanticModel doesn't bind an expression body in presence of a block body." What specific API we will use for that purpose is not important right now, we are not writing the tests.

Reply:
[https://github.com/dotnet/roslyn/pull/57076#discussion_r775257197](https://github.com/dotnet/roslyn/pull/57076#discussion_r775257197) Should provide some hints if you are actually trying to create a test.